### PR TITLE
fix: core lib update for various custom operator conformance fixes

### LINKED ISSUE #867: Update flagd-testbed to v3.6.0 and fix in-process evaluation edge cases
## Summary

Update the flagd-testbed dependency to [v3.6.0](https://github.com/open-feature/flagd-testbed/releases/tag/v3.6.0) and ensure the flagd provider passes all new edge-case tests.

Testbed v3.6.0 adds coverage for the edge cases identified in:
- open-feature/flagd#1873 (sem_ver parsing inconsistencies)
- open-feature/flagd#1874 (custom operator error return values)
- open-feature/flagd#1875 ($evaluators/$ref resolution gaps)

## What's needed

1. **Update the flagd evaluation library dependency** to pick up fixes from open-feature/flagd
2. **Update flagd-testbed to v3.6.0**
3. **Verify all new edge-case tests pass**; the underlying bugs are fixed in the flagd evaluation library, so this is primarily a dependency update

## Bugs fixed transitively (via flagd library update)

1. sem_ver returns `false` on error instead of `nil`
2. `ends_with` returns `false` on error while `starts_with` returns `nil`
3. `fractional` returns `""` on no-bucket-match instead of `nil`
4. Non-deterministic `$evaluators` replacement ordering
5. No diagnostic feedback on unresolved `$ref`s at parse time

Parent: open-feature/flagd#1904

### PR DESCRIPTION
**Pull request by toddbaert**
* support V prefix in semver
* return null on errors in semver, ends_with, and fractional (already existed on starts_with)
* fix fractional fallback when total weight is zero

Done by updating flagd-core module and test-harness, removing some test exclusions.

Fixes: https://github.com/open-feature/go-sdk-contrib/issues/867

---

Related:
* https://github.com/open-feature/flagd/pull/1950
* https://github.com/open-feature/java-sdk-contrib/pull/1778
* https://github.com/open-feature/js-sdk-contrib/pull/1519
* https://github.com/open-feature/dotnet-sdk-contrib/pull/635
* https://github.com/open-feature/python-sdk-contrib/pull/386

### CODE REVIEW

**Review by beeme1mr — APPROVED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -4,4 +4,3 @@
 [submodule "providers/flagd/flagd-testbed"]
 	path = providers/flagd/flagd-testbed
 	url = https://github.com/open-feature/flagd-testbed.git
-	branch = v2.11.1
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply .gitmodules:7]: yes. we want this,
